src/modules/textFacets/__tests__/textFacets.test.js

```diff
@@ -0,0 +1,127 @@
+    window.fetch = jest.fn((request) => {
+        if (request.includes('filter=brand_uFilter:"Scarpa"')) {
+            return Promise.resolve({
+                json: () => Promise.resolve(facetResponse),
+            });
+        }
+        return Promise.resolve({
+            json: () => Promise.resolve(searchResponse),
+        });
+    });
+})
+
+const attributesMap = {
+    productName: 'title',
+    uniqueId: 'uniqueId',
+    imageUrl: 'imageUrl',
+    price: 'RRP_Price',
+    productUrl: 'productUrl'
+};
+
+
+test('Match Snapshot for text facets', async () => {
+    const tree = renderer
+        .create(
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >
+                <TextFacets />
+                <Products attributesMap={attributesMap} />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+
+        );
+        await waitFor(() => tree.toJSON());
+        expect(tree.toJSON()).toMatchSnapshot();
+});
+
+test('Test text facet click', async () => {
+    const { getByText } = render(
+        <>
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >   
+                <TextFacets />
+                <Products
+                    attributesMap={attributesMap}
+                />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa - 18')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa - 18"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
+        fireEvent.click(getByText("Clear"));
+    });
+});
+
+test('Test text facet click with FacetItemComponent', async () => {
+    const { getByText } = render(
+        <>
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >   
+                <TextFacets facetItemComponent={<FacetItemComponent />}/>
+                <Products
+                    attributesMap={attributesMap}
+                />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
```

Comment:
Please check the customizations everywhere.

Reply:
done